icx-proxy/src/main.rs

```diff
@@ -465,7 +461,9 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     let counter = AtomicUsize::new(0);
     let debug = opts.debug;
-    let check_uri_and_referer_for_canister_id = opts.check_uri_and_referer_for_canister_id;
+
+    /// For local/development, check the URI (query string) and the referer for a canister id.
+    let check_uri_and_referer_for_canister_id = opts.address.ip().is_loopback();
```

Comment:
Nice in principle. But does that work? Are we sure that the deployed icx-proxy doesn’t sit behind some form of proxy _on the same machine_?

Reply:
Yeah, it does not work, both because of that and because in production icx-proxy binds to localhost, behind nginx

Reply:
Yes, that’s what what I mean. Too bad.

Reply:
Hh yes that is exactly what you wrote